### PR TITLE
doc/release-note-3.2: MCUboot changeloog

### PR DESCRIPTION
**Pull request by nvlsianpu**
Added summary of MCUboot changes for 3.2 release

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

**Review by fabiobaltieri — DISMISSED** [review]

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -955,6 +955,28 @@ HALs
...
+  * Added initial support for leveraging the RAM-LOAD mode with the zephyr-rtos port.
+  * Added the MCUboot status callback support.
+    See :kconfig:option:`CONFIG_MCUBOOT_ACTION_HOOKS`.
+  * Edited includes to have the ``zephyr/`` prefix.
+  * Edited the DFU detection's GPIO-pin configuration to be done through DTS using the ``mcuboot-button0`` pin alias.
+  * Edited the LED usage to prefer DTS' ``mcuboot-led0`` alias over the ``bootloader-led0`` alias.
+  * Removed :c:func:`device_get_binding()` usage in favor of :c:func:`DEVICE_DT_GET()`.
+  * Added support for generic `watchdog0` alias.
+  * Enabled watchdog feed by default.
+  * Droped :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOKS_FILE` option. Inclusion of Hooks implementation file is now up to the projects customization.
```
Comment (nordicjm) [inline doc/releases/release-notes-3.2.rst:967]: *Dropped

File: doc/releases/release-notes-3.2.rst [hunk 2]
```diff
@@ -955,6 +955,28 @@ HALs
...
+  * Switched zephyr port from using FLASH_AREA_ macros to FIXED_PARTITION_ macros
+  * Made flash_map_backend.h compatible with a C++ compiler
+  * Allowed to get the flash write alignment basing on the zephyr,flash DT chosen node property
```
Comment (nordicjm) [inline doc/releases/release-notes-3.2.rst:970]: *based

**Review by nordicjm — COMMENTED** [review]
Typo in commit: *changelog

**Review by nordicjm — DISMISSED** [review]

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
CI is not happy: `/home/runner/work/zephyr/zephyr/doc/_build/src/releases/release-notes-3.2.rst:968:Unknown target name: "flash_area".`

File: doc/releases/release-notes-3.2.rst [hunk 3]
```diff
@@ -955,6 +955,28 @@ HALs
...
+  * Added the MCUboot status callback support.
+    See :kconfig:option:`CONFIG_MCUBOOT_ACTION_HOOKS`.
+  * Edited includes to have the ``zephyr/`` prefix.
+  * Edited the DFU detection's GPIO-pin configuration to be done through DTS using the ``mcuboot-button0`` pin alias.
+  * Edited the LED usage to prefer DTS' ``mcuboot-led0`` alias over the ``bootloader-led0`` alias.
+  * Removed :c:func:`device_get_binding()` usage in favor of :c:func:`DEVICE_DT_GET()`.
+  * Added support for generic `watchdog0` alias.
+  * Enabled watchdog feed by default.
+  * Dropped :kconfig:option:`CONFIG_BOOT_IMAGE_ACCESS_HOOKS_FILE` option. Inclusion of Hooks implementation file is now up to the projects customization.
+  * Switched zephyr port from using FLASH_AREA_ macros to FIXED_PARTITION_ macros
```
Comment (fabiobaltieri) [inline doc/releases/release-notes-3.2.rst:968]: I think FLASH_AREA_ and FIXED_PARTITION_ should be escaped.

**Review by fabiobaltieri — DISMISSED** [review]

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Just a pair of suggestions

**Review by FrancescoSer — DISMISSED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Review by FrancescoSer — APPROVED** [review]

**Pull request merged**